opencog/cython/opencog/utilities.pyx

```diff
@@ -1,43 +1,54 @@
 from contextlib import contextmanager
 from opencog.atomspace cimport AtomSpace
 from opencog.atomspace import create_child_atomspace
-from opencog.type_constructors import get_type_ctor_atomspace, set_type_ctor_atomspace
+from opencog.type_constructors import get_default_atomspace, set_default_atomspace
+import warnings
+
 
 # Avoid recursive intialization
 is_initialized = False
 
-def initialize_opencog(AtomSpace atomspace):
 
+def initialize_opencog(AtomSpace atomspace=None):
```

Comment:
Actually I am not sure if we should deprecate that. Sometimes it is better to make initialization manually to know exact moment when it is initialized instead of initializing it in some undefined moment.

Reply:
initialize_opencog is not deprecated, but i want to deprecate passing an atomspace to initialize_opencog

Reply:
But `initialize_opencog` isn't needed any more: it can be done automatically, at the time that `import opencog` is done. 

The only reason it was exported is because it needed the AtomSpace argument, and the AtomSpace was unknown at the time of `import`. Now that it doesn't need that argument, it can be moved back to the `import`.

Reply:
The be clear: the call `c_initialize_python()` should be moved to the import -- If I recall correctly, it would be moved to `__init__.py`  and then get rid if the `global is_initialized` flag, its not needed.  All that's left is that `initialize_opencog()` is another deprecated synonym for `set_default_atomspace()`.  Meanwhile, `c_finalize_python()` would get called from the python finalizer (e.g. in `__init__.py`, you'd have some class with a `__del__` method on it, and it would call `c_finalize_python()` That way, when python exits, the class gets deleted, `__del__` gets called.